### DISCUSSION TITLE
Difference between Meta's MMM and Google's Meridian

### DISCUSSION THREAD

**Discussion by OmkarEllicium** [google/meridian#673] [General]
Hi Team,

Can anyone please help with the major differences between Meta's Robyn for MMM and Google's Meridian?
Mostly I would like to know what happens differently inside Meridian? Any major algorithm differences between these two?

**Comment by kgierach** [comment]
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN" "http://www.w3.org/TR/html4/strict.dtd">
<html>
<head>
<meta http-equiv="Content-Style-Type" content="text/css">
<meta name="Generator" content="Cocoa HTML Writer">
<meta name="CocoaVersion" content="2113.65">
</head>
<body>
<p class="p1">Google's Meridian and Meta's Robyn are two leading Marketing Mix Modeling (MMM) tools with distinct technical approaches and use-case specializations. Here's a detailed comparison of their capabilities and algorithms:</p>
<p class="p2"><span class="Apple-converted-space">&nbsp;</span></p>
<p class="p3"><b>Core Methodologies</b></p>
<p class="p1"><b>Meridian</b></p>
<ul class="ul1">
<li class="li4"><span class="s1"></span>Uses <b>Bayesian causal inference</b> to propagate uncertainty across parameter estimates<span class="s2"><sup></sup></span></li>
<li class="li4"><span class="s1"></span>Implements <b>hierarchical geo-level modeling</b> to analyze regional performance variations<span class="s2"><sup></sup></span></li>
<li class="li4"><span class="s1"></span>Incorporates ROI priors from experiments/benchmarks as Bayesian parameters<span class="s2"><sup></sup></span></li>
</ul>
<p class="p1"><b>Robyn</b></p>
<ul class="ul1">
<li class="li4"><span class="s1"></span>Combines <b>Ridge regression</b> with <b>evolutionary algorithms</b> (TwoPointsDE) for hyperparameter optimization<span class="s2"><sup></sup></span></li>
<li class="li4"><span class="s1"></span>Employs <b>multi-objective optimization</b> balancing prediction accuracy (NRMSE) and business metrics (DECOMP.RSSD)<span class="s2"><sup></sup></span></li>
<li class="li1"><span class="s1"></span>Uses Facebook Prophet for <b>time-series decomposition</b> of trends/seasonality<span class="s2"><sup></sup></span></li>
</ul>
<p class="p2"><span class="Apple-converted-space">&nbsp;</span></p>
<p class="p5"><b>Key Capabilities Comparison</b></p>

Feature | Meridian | Robyn
-- | -- | --
Data Handling | Optimized for geo-level data (50+ regions) | Built for high-dimensional digital datasets
Uncertainty Modeling | Full Bayesian posterior distributions | SHAP values for explainability
Media Impact Models | Reach/frequency-based video analysis | Adstock & saturation curves (geometric/Weibull)
Customization | Open-source TensorFlow Probability | Modular Python implementation
Optimization Focus | Maximizing media ROI with priors | Balancing accuracy vs. business logic


<p class="p8"><br></p>
<p class="p2"><span class="Apple-converted-space">&nbsp;</span></p>
<p class="p3"><b>Algorithmic Differences</b></p>
<p class="p1"><b>Meridian's Bayesian Engine</b></p>
<ul class="ul1">
<li class="li4"><span class="s1"></span>Uses Markov Chain Monte Carlo (MCMC) sampling for posterior inference<span class="s2"><sup></sup></span></li>
<li class="li4"><span class="s1"></span>Models time-varying intercepts for organic growth/seasonality<span class="s2"><sup></sup></span></li>
<li class="li4"><span class="s1"></span>Integrates Google Search query volume to reduce endogeneity bias<span class="s2"><sup></sup></span></li>
</ul>
<p class="p1"><b>Robyn's ML Pipeline</b></p>
<ol class="ol1">
<li class="li4"><span class="s4"></span><b>Regularization</b>: Ridge regression handles multicollinearity<span class="s2"><sup></sup></span></li>
<li class="li4"><span class="s4"></span><b>Evolutionary Optimization</b>: Nevergrad's TwoPointsDE algorithm<span class="s2"><sup></sup></span></li>
<li class="li4"><span class="s4"></span><b>Validation</b>: Time-series split with train/validation/test sets<span class="s2"><sup></sup></span></li>
<li class="li1"><span class="s4"></span><b>Clustering</b>: Groups similar media channels for strategy insights<span class="s2"><sup></sup></span></li>
</ol>
<p class="p2"><span class="Apple-converted-space">&nbsp;</span></p>
<p class="p3"><b>Ideal Use Cases</b></p>
<p class="p1"><b>Choose Meridian When:</b></p>
<ul class="ul1">
<li class="li4"><span class="s1"></span>You need geo-level performance insights</li>
<li class="li4"><span class="s1"></span>Prior experimental data exists for Bayesian priors</li>
<li class="li4"><span class="s1"></span>Transparency in uncertainty estimates is critical<span class="s2"><sup></sup></span></li>
</ul>
<p class="p1"><b>Choose Robyn When:</b></p>
<ul class="ul1">
<li class="li4"><span class="s1"></span>Working with granular digital marketing data</li>
<li class="li4"><span class="s1"></span>Automated hyperparameter tuning is preferred</li>
<li class="li4"><span class="s1"></span>Balancing statistical and business metrics is key<span class="s2"></span></li>
</ul>
<p class="p1">Both tools represent modern MMM evolution - Meridian through Bayesian causal frameworks, and Robyn via machine learning automation. The choice depends on data structure, technical resources, and specific measurement priorities.</p>
<p class="p1">I hope that helps as a high level overview, but I'm happy to dive deeper into more specifics if that would help.</p>

</body>
</html>

**Comment by dattatreyam23** [comment]
Hi @OmkarEllicium ,

Thank you for contacting us!

Meridian uses Bayesian inference with an emphasis on causality (see [Rationale for causal inference and Bayesian modeling](https://developers.google.com/meridian/docs/basics/rationale-for-causal-inference-and-bayesian-modeling)). Robyn uses ridge regression and multi-objective optimization for hyperparameter tuning and model calibration.

Meridian’s has a few key innovations, including the integration of richer data signals like search query volume and reach and frequency metrics ([Zhang et al., 2023](https://research.google/pubs/bayesian-hierarchical-media-mix-model-incorporating-reach-and-frequency-data/)), and improved methods for model calibration using experimental results ([Zhang et al., 2024](https://research.google/pubs/media-mix-model-calibration-with-bayesian-priors/)).

Feel free to reach out to us in case you have any other queries regarding Meridian.

Thank you,

Google Meridian Support Team

**Comment by OmkarEllicium** [comment]
Thanks @dattatreyam23 @kgierach for the information. One observation I have seen, for the media spend channels, the contribution is more in Meridian as compared to Robyn. Is there any specific reason for this?

**Reply by dattatreyam23** [reply]
Hi  @OmkarEllicium ,

Differences in contribution results often arise from a combination of factors. Both the specific characteristics and granularity of the input data, and the distinct underlying methodologies used by each solution, play a role. Together, these aspects can naturally lead to differences between the two approaches.

Thank you,

Google Meridian Support Team
